### ISSUE TITLE
defalut config ineffectiveness

### ISSUE THREAD

**Issue by ApplePieC** [acouvreur/sablier#136]
```
http:
  services:
    pigallery2-service:
      loadBalancer:
        servers:
          - url: "http://pigallery2:80"

  routers:
    pigallery2-router:
      rule: "Host(`abcdefg.com`)"
      service: pigallery2-service
      middlewares:
       - pigallery2-sablier

  middlewares:
    pigallery2-sablier:
      plugin:
        sablier:
          blocking:
            timeout: 1m
          names: pigallery2
          sablierUrl: http://sablier:10000
          sessionDuration: 1m
```
It works fine.
However, removing timeout or sessionDuration will show an error.
I would like to be able to remove these unnecessary parameters to apply the default configuration.

**Comment by acouvreur** [comment]
That makes sense!
We have to update the plugins configuration creation

**Comment by ApplePieC** [comment]
> That makes sense! We have to update the plugins configuration creation

refreshFrequency is also invalid.I set it to 1s, 2s, 3s, and it didn't work.It will only be maintained at 5s.

In addition, I asked if if I set up multiple middleware for one middleware to shut down the container and affect the use of other containers if one middleware is not listening for requests. For example, I use both blocking and dynamic effects to point to the same container, and after dyanmic's access is canceled, the blocking is still accessing, will the container be closed?

**Comment by ApplePieC** [comment]
> That makes sense! We have to update the plugins configuration creation

Also, what technology does sablier use to listen to the status of the container and shut down the container. Whether to scan at intervals or set a delay in closing at the last response？

**Comment by ApplePieC** [comment]
> That makes sense! We have to update the plugins configuration creation

Or what exactly storage is to save, and what will be the impact of me not setting it.

**Comment by acouvreur** [comment]
> > That makes sense! We have to update the plugins configuration creation
> 
> Or what exactly storage is to save, and what will be the impact of me not setting it.

If the service had setup some service to stop in one hour, and Sablier shuts down for any reason, it will be picked up again at the next start up. I'll add this in the documentation

**Comment by acouvreur** [comment]
> > That makes sense! We have to update the plugins configuration creation
> 
> refreshFrequency is also invalid.I set it to 1s, 2s, 3s, and it didn't work.It will only be maintained at 5s.
> 
> In addition, I asked if if I set up multiple middleware for one middleware to shut down the container and affect the use of other containers if one middleware is not listening for requests. For example, I use both blocking and dynamic effects to point to the same container, and after dyanmic's access is canceled, the blocking is still accessing, will the container be closed?

Containers status are stored by their name, and it stores at which time it should be evicted, so it's the last access by middleware with its own session duration on it.

is it clear?

**Comment by ApplePieC** [comment]
> > > That makes sense! We have to update the plugins configuration creation
> > 
> > 
> > refreshFrequency is also invalid.I set it to 1s, 2s, 3s, and it didn't work.It will only be maintained at 5s.
> > In addition, I asked if if I set up multiple middleware for one middleware to shut down the container and affect the use of other containers if one middleware is not listening for requests. For example, I use both blocking and dynamic effects to point to the same container, and after dyanmic's access is canceled, the blocking is still accessing, will the container be closed?
> 
> Containers status are stored by their name, and it stores at which time it should be evicted, so it's the last access by middleware with its own session duration on it.
> 
> is it clear?

In this case, can you add a function. When the Sablier storage container should be closed, it can only cover the earlier time. This can achieve an effect. When multiple middleware access to the same container, the container will not be interrupted by accident, which is logical.

**Comment by acouvreur** [comment]
Can you provide more info about your setup?
How is the Sablier default configuration set?

**Comment by ApplePieC** [comment]
> 您能提供有关您的设置的更多信息吗？如何设置Sablier默认配置？

```
provider:
  # Provider to use to manage containers (docker, swarm, kubernetes)
  name: docker 
server:
  # The server port to use
  port: 10000 
  # The base path for the API
  base-path: /
storage:
 # File path to save the state (default stateless)
 file:
sessions:
  # The default session duration (default 5m)
  default-duration: 5m
  # The expiration checking interval. 
  # Higher duration gives less stress on CPU. 
  # If you only use sessions of 1h, setting this to 5m is a good trade-off.
  expiration-interval: 20s
logging:
  level: trace
strategy:
  dynamic:
    # Custom themes folder, will load all .html files recursively (default empty)
    custom-themes-path:
    # Show instances details by default in waiting UI
    show-details-by-default: false
    # Default theme used for dynamic strategy (default "hacker-terminal")
    default-theme: hacker-terminal
    # Default refresh frequency in the HTML page for dynamic strategy
    default-refresh-frequency: 5s
  blocking:
    # Default timeout used for blocking strategy (default 1m)
    default-timeout: 1m
```
```
# photoprism      2342   dynamic
# pigallery2      80     blocking
# pigallery2-2    80     blocking
# bookbrowser     8090   dynamic
# prowlarr        9696   blocking
# chatgpt         3002   dynamic
# chatgptapi      3002   dynamic
http:
  services:
    photoprism-service:
      loadBalancer:
        servers:
          - url: http://photoprism:2342
    pigallery2-service:
      loadBalancer:
        servers:
          - url: http://pigallery2:80
    pigallery2-2-service:
      loadBalancer:
        servers:
          - url: http://pigallery2-2:80
    bookbrowser-service:
      loadBalancer:
        servers:
          - url: http://bookbrowser:8090
    prowlarr-service:
      loadBalancer:
        servers:
          - url: http://prowlarr:9696
    chatgpt-service:
      loadBalancer:
        servers:
          - url: http://chatgpt:3002
    chatgptapi-service:
      loadBalancer:
        servers:
          - url: http://chatgptapi:3002
  routers:
    photoprism-router:
      rule: Host(`photoprism-b.example.com`)
      service: photoprism-service
      middlewares:
        - photoprism-sablier
    pigallery2-router:
      rule: Host(`pigallery2-b.example.com`)
      service: pigallery2-service
      middlewares:
        - pigallery2-sablier
    pigallery2-2-router:
      rule: Host(`pigallery2-2-b.example.com`)
      service: pigallery2-2-service
      middlewares:
        - pigallery2-2-sablier
    bookbrowser-router:
      rule: Host(`bookbrowser-b.example.com`)
      service: bookbrowser-service
      middlewares:
        - bookbrowser-sablier
    prowlarr-router:
      rule: Host(`prowlarr-b.example.com`)
      service: prowlarr-service
      middlewares:
        - prowlarr-sablier
    chatgpt-router:
      rule: Host(`chatgpt-b.example.com`)
      service: chatgpt-service
      middlewares:
        - chatgpt-sablier
    chatgptapi-router:
      rule: Host(`chatgptapi-b.example.com`)
      service: chatgptapi-service
      middlewares:
        - chatgptapi-sablier
  middlewares:
    photoprism-sablier:
      plugin:
        sablier:
          names: photoprism
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          dynamic:
            displayName: photoprism
            showDetails: true
            theme: shuffle
            refreshFrequency: 5s
    pigallery2-sablier:
      plugin:
        sablier:
          names: pigallery2
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          blocking:
            timeout: 1m
    pigallery2-2-sablier:
      plugin:
        sablier:
          names: pigallery2-2
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          blocking:
            timeout: 1m
    bookbrowser-sablier:
      plugin:
        sablier:
          names: bookbrowser
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          dynamic:
            displayName: bookbrowser
            showDetails: true
            theme: shuffle
            refreshFrequency: 5s
    prowlarr-sablier:
      plugin:
        sablier:
          names: prowlarr
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          blocking:
            timeout: 1m
    chatgpt-sablier:
      plugin:
        sablier:
          names: chatgpt
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          dynamic:
            displayName: chatgpt
            showDetails: true
            theme: shuffle
            refreshFrequency: 5s
    chatgptapi-sablier:
      plugin:
        sablier:
          names: chatgptapi
          sablierUrl: http://sablier:10000
          sessionDuration: 5m
          dynamic:
            displayName: chatgptapi
            showDetails: true
            theme: shuffle
            refreshFrequency: 5s
```
Worried about the problem of accidental interruption of the container to the same container by the same container. I am currently a single middleware pointing to a single container. So ask you the above questions, hoping to get answers.

**Comment by ApplePieC** [comment]
> 🎉 This issue has been resolved in version 1.3.1-beta.1 🎉
> 
> The release is available on:
> 
> * `v1.3.1-beta.1`
> * [GitHub release](https://github.com/acouvreur/sablier/releases/tag/v1.3.1-beta.1)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

Hello, I read the code about my questions. My above problems have been solved. Thank you for your efforts.
